File: src/data/console/commands/niko_MPC_loadSettings.java
package data.console.commands;

import data.niko_MPC_modPlugin;
import data.utilities.niko_MPC_ids;
import org.jetbrains.annotations.NotNull;
import org.json.JSONException;
import org.lazywizard.console.BaseCommand;

import java.io.IOException;

public class niko_MPC_loadSettings implements BaseCommand {
    @Override
    public CommandResult runCommand(@NotNull String args, @NotNull CommandContext context) {
        try {
            niko_MPC_modPlugin.loadSettings();
        } catch (IOException | JSONException | NullPointerException ex) {
            throw new RuntimeException(niko_MPC_ids.niko_MPC_masterConfig + " loading failed during command execution!");
        }
        return CommandResult.SUCCESS;
    }
}

File: src/data/console/commands/niko_MPC_loadSettings.kt
package data.console.commands

import com.fs.starfarer.api.Global
import com.fs.starfarer.api.campaign.CampaignTerrainAPI
import com.fs.starfarer.api.campaign.StarSystemAPI
import data.utilities.niko_MPC_settings.loadAllSettings
import data.utilities.niko_MPC_settings.loadSettings
import org.apache.log4j.Level
import org.lazywizard.console.BaseCommand
import org.lazywizard.console.BaseCommand.CommandContext
import org.lazywizard.console.BaseCommand.CommandResult
import org.lazywizard.console.Console

class niko_MPC_loadSettings : BaseCommand {
    override fun runCommand(args: String, context: CommandContext): CommandResult {
        try {
            loadAllSettings()
        } catch (ex: Exception) {
            val errorCode = "runCommand failed due to thrown exception: $ex, ${ex.cause}"
            Console.showMessage(errorCode, Level.ERROR)
            return CommandResult.ERROR
        }
        Console.showMessage("Success! Settings have been reloaded.")

        return CommandResult.SUCCESS
    }
}